File: examples/swift-client/Sources/AppwriteModels/CollectionList.swift

/// Collections List
public class CollectionList {

    /// Total number of items available on the server.
    public let sum: Int

    /// List of collections.
    public let collections: [Collection]

    init(
        sum: Int,
        collections: [Collection]
    ) {
        self.sum = sum
        self.collections = collections
    }

    public static func from(map: [String: Any]) -> CollectionList {
        return CollectionList(
            sum: map["sum"] as! Int,
            collections: (map["collections"] as! [[String: Any]]).map { Collection.from(map: $0) }
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "sum": sum as Any,
            "collections": collections.map { $0.toMap() } as Any
        ]
    }
                                                                                                                                                                    
}

File: examples/flutter/lib/src/models/collection_list.dart
part of appwrite.models;

/// Collections List
class CollectionList {
    /// Total number of items available on the server.
    final int sum;
    /// List of collections.
    final List<Collection> collections;

    CollectionList({
        required this.sum,
        required this.collections,
    });

    factory CollectionList.fromMap(Map<String, dynamic> map) {
        return CollectionList(
            sum: map['sum'],
            collections: List<Collection>.from(map['collections'].map((p) => Collection.fromMap(p))),
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "sum": sum,
            "collections": collections.map((p) => p.toMap()),
        };
    }
}
